decoder.py

```diff
@@ -55,12 +55,16 @@ def decode_gif(io: typing.BinaryIO) -> Gif:
 
             decode_image_data(gif_stream, gif_object)
 
-            if gif_object.graphic_control_extensions[
-                gif_object.images[LAST_ELEMENT].index_graphic_control_ex].disposal == 3:
+            last_graphic_control_index = gif_object.images[LAST_ELEMENT].index_graphic_control_ex
+            last_graphic_control_disposal = gif_object.graphic_control_extensions[last_graphic_control_index].disposal
+            if last_graphic_control_disposal == 3:
                 gif_object.images.append(gif_object.images[PENULTIMATE])
+            elif last_graphic_control_disposal == 2:
+                print("has disposal 2, need to check if correct")
+                background_frame(gif_object)
```

Comment:
we cant add with no example for test..pls find an example(maybe u can create one)

Reply:
i will try
also, for some reason some other changes from other branches made their way to this branch, i will recreate the branch and will pr again.